app/models/event.rb

```diff
@@ -154,6 +154,17 @@ def compute_occupied_places
     application_letters.where(status: ApplicationLetter.statuses[:accepted]).count
   end
 
+  # locks the ability to change application statuses
+  def lock_application_status
+    application_status_locked = true
+    update(application_status_locked: true)
+  end
+  # unlocks the ability to change application statuses
+  def unlock_application_status
```

Comment:
Maybe you could outsource it to the tests? I dont think having code only needed for tests in productive model code is good style